kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/admin/ProcessInstanceMigrationIntegrationTest.java

```diff
@@ -266,4 +284,92 @@ public void testUpgradeProcessInstancesWithNodeMapping() throws Exception {
             }
         }
     }
+
+
+    @Test
+    public void testUpgradeParentInstanceWithIndependentSubprocess() {
+        Long processParentInstanceId = processClient.startProcess(CONTAINER_ID, PARENT_WITH_NON_INDEPENDENT_SUBPROCESS_ID);
+        assertNotNull(processParentInstanceId);
+        assertTrue(processParentInstanceId > 0);
+
+        try {
+            logger.info("Process in container {} has started {}", CONTAINER_ID, processParentInstanceId);
+            List<ProcessInstance> childrenProcessInstances = processClient.findProcessInstancesByParent(CONTAINER_ID, processParentInstanceId, 0, 10);
+            assertEquals(2, childrenProcessInstances.size());
+            for (ProcessInstance processInstance : childrenProcessInstances) {
+                logger.info("Subrocess in container {} has started {}", CONTAINER_ID, processInstance.getId());
+                assertEquals("1.0", processInstance.getProcessVersion());
+                assertEquals(CONTAINER_ID, processInstance.getContainerId());
+                assertEquals(SUBPROCESS_ID, processInstance.getProcessId());
+                assertEquals(processParentInstanceId, processInstance.getParentId());
+            }
+
+            List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertEquals(2, tasks.size());
+            for (TaskSummary task : tasks) {
+                assertEquals("Evaluate items?", task.getName());
+                assertEquals(CONTAINER_ID, task.getContainerId());
+                assertEquals(SUBPROCESS_ID, task.getProcessId());
+            }
+
+            MigrationSpecification spec = new MigrationSpecification();
+            MigrationProcessSpecification pSpecParent = new MigrationProcessSpecification();
+            pSpecParent.setSourceProcessId(PARENT_WITH_INDEPENDENT_SUBPROCESS_ID);
```

Comment:
Source process Id is `PARENT_WITH_NON_INDEPENDENT_SUBPROCESS_ID`
```suggestion
            pSpecParent.setSourceProcessId(PARENT_WITH_NON_INDEPENDENT_SUBPROCESS_ID);
```

Reply:
nope